test/parallel/test-linux-dev-stdin.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+const common = require('../common');
+
+//
+// This test ensures Node.js doesn't crash on linux when reading from /dev/stdin in the CLI
+// ref: https://github.com/nodejs/node/issues/54200
+//
+
+if (!common.isLinux) {
+  common.skip('Linux only test');
+}
+
+const child_process = require('child_process');
+const assert = require('assert');
+const fs = require('fs');
+
+const cp = child_process.execSync(`printf 'console.log(1)' | "${process.execPath}" /dev/stdin`, { stdio: 'pipe' });
+assert.strictEqual(cp.toString(), '1\n');
+
+if (fs.existsSync('/tmp/stdin')) {
+  fs.unlinkSync('/tmp/stdin');
+}
+fs.symlinkSync('/dev/stdin', '/tmp/stdin');
```

Comment:
Can you add a test for the fd 0 symlink and just /dev/stdin without symlinks)

(I don't remember what the fd 0 symlink is off the top of my head, but `readlink /dev/stdin` will tell you)

You can use `node:test` to split it up